File: C_plus_plus/Merge_Sorted_Array.cpp
class Solution {
public:
    void merge(int A[], int m, int B[], int n) {
        for(int i = m-1 ; i >= 0 ; --i)
            A[n+i] = A[i];
        int i = 0, j = 0, k = 0;
        while(i < m && j < n){
            if(A[n+i] < B[j]) A[k++] = A[n+i++];
            else A[k++] = B[j++];
        }
        while(i<m) A[k++] = A[n+i++];
        while(j<n) A[k++] = B[j++];
    }
};

File: java/Merge_Sorted_Array.java
// Fill the number at the end of the array
// time is O(n), space is O(1).

public class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int a = m-1, b = n-1;
        for (int i = m+n-1; i >= 0; --i) {
            if (b < 0 || a >=0 && nums1[a] >= nums2[b])
                nums1[i] = nums1[a--];
            else nums1[i] = nums2[b--];
        }
    }
}
